packages/backend-common/src/service/lib/config.ts

```diff
@@ -79,6 +97,38 @@ export function readCorsOptions(config: ConfigReader): CorsOptions | undefined {
   });
 }
 
+/**
+ * Attempts to read a certificate options object from the root of a config object.
+ *
+ * @param config The root of a backend config object
+ * @returns A certificate options object, or undefined if not specified
+ *
+ * @example
+ * ```json
+ * {
+ *   certificate: {
+ *    key: ...,
+ *    attributes: ...
+ *   }
+ * }
+ * ```
+ */
+export function readCertificateOptions(
+  config: ConfigReader,
+): CertificateOptions | undefined {
+  const cc = config.getOptionalConfig('certificate');
+  if (!cc) {
+    return undefined;
+  }
+
+  return removeUnknown({
+    key: cc.getOptionalConfig('key') as CertificateKeyOptions,
```

Comment:
`get*Config` returns a narrower view of the config, so you still have to use the config API to read out values here. That is the returned value won't match `CertificateKeyOptions`. Typescript is only ok with this cast since it's guaranteed to be an object and all of the props in the target type are optional.

Reply:
Could you possibly elaborate on this with an example? I was kinda hoping I wouldnt have to process all the sub-sections "member-by-member" sort of speak. :P